src/api.v2/middlewares/authMiddlewares.js

```diff
@@ -30,7 +32,17 @@ const checkUserAuthenticated = (req, next) => {
 
   return true;
 };
+// Throws if the user hasnt agreed to the privacy policy yet
+const checkForPrivacyPolicyAgreement = (req, next) => {
+  const isBiomageDeployment = BIOMAGE_DOMAIN_NAMES.includes(config.domainName);
 
+  if (req.user['custom:agreed_terms'] !== 'true' && isBiomageDeployment) {
+    next(new NotAgreedToTermsError('The user hasnt agreed to the privacy policy yet.'));
+    return false;
+  }
+
+  return true;
+};
```

Comment:
this could be a candidate for another domain specific content as it is something that is only used for the Biomage deployment